tiledb/sm/array_schema/test/unit_dimension.cc

```diff
@@ -34,8 +34,105 @@
 #include "../dimension.h"
 #include "tiledb/sm/enums/datatype.h"
 
+using namespace tiledb;
+using namespace tiledb::common;
 using namespace tiledb::sm;
 
+template <class T, int n>
+inline T& dim_buffer_offset(void* p) {
+  return *static_cast<T*>(static_cast<void*>(static_cast<char*>(p) + n));
+}
+
 TEST_CASE("Dimension::Dimension") {
   Dimension x{"", Datatype::UINT32};
 }
+
+TEST_CASE("Dimension: Test deserialize,int32", "[dimension][deserialize]") {
+  char serialized_buffer[40];
+  char* p = &serialized_buffer[0];
+  uint32_t dimension_name_size = 2;
+  std::string dimension_name = "d1";
+  dim_buffer_offset<uint32_t, 0>(p) = dimension_name_size;
+  std::memcpy(
+      &dim_buffer_offset<char, 4>(p),
+      dimension_name.c_str(),
+      dimension_name_size);
+
+  uint8_t datatype = (uint8_t)Datatype::INT32;
+  dim_buffer_offset<uint8_t, 6>(p) = datatype;
```

Comment:
These tests will be much easier to read if all the data members were in one sequence and then all the offset assignments in a second block. The interleaved order here and below makes it harder to focus on the different concerns.